sherpa/astro/tests/test_astro_pha_scale.py

```diff
@@ -0,0 +1,1191 @@
+#
+#  Copyright (C) 2017  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""
+Test handling of area/background scaling in PHA data sets.
+
+A PHA file can have AREASCAL and BACKSCAL values (either a scalar or
+column). The BACKSCAL column has tests in other parts of the system,
+so this file concentrates on the area scaling column.
+
+The handling of AREASCAL is scattered throughout the system: it's
+not just the DataPHA object but also the ARF/RMF/RSP PHA instrument
+objects.
+
+The tests here focus on the object API, rather than the UI layer
+(which, at present, doesn't do much extra with regard to area
+scaling, so this should catch most problems).
+
+This does *not* test any plots (e.g. to ensure that the correctly-scaled
+data and/or model values are shown), but does test some of the accessor
+methods the plotting code uses (e.g. get_y).
+
+The unit-level tests here (mostly) focus on just the area scaling; the
+more integration-level tests (i.e. those that read in data files)
+are used to check that the scaling is being applied to cases
+where the exposure time or background scaling may not be unity.
+
+"""
+
+import pytest
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from sherpa.astro.data import DataARF, DataPHA, DataRMF
+from sherpa.astro.instrument import ARFModelPHA, RMFModelPHA, RSPModelPHA
+from sherpa.models.basic import Const1D, StepHi1D
+from sherpa.stats import Chi2DataVar, CStat
+from sherpa.utils.err import ArgumentErr, DataErr
+
+from sherpa.utils import requires_data, requires_fits
+
+from sherpa.astro import ui
+
+
+# pytest.mark.parametrize and requires_data do not seem to be playing
```

Comment:
I have limited time to make any significant changes.

I'm not 100% convinced whether the diagnosis of the problem I made is correct (it seems to be, but allI did was fins what appeared to be a workable workaround, rather than to diagnose the problem(.